src/formats/cif.cpp

```diff
@@ -0,0 +1,205 @@
+//
+// Created by danny305 on May 3rd, 2021.
+//
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <stdexcept>
+#include <fstream>
+#include <filesystem>
+
+#include <gemmi/cif.hpp>
+#include <gemmi/mmcif.hpp>
+
+#define GEMMI_WRITE_IMPLEMENTATION
+#include <gemmi/to_cif.hpp>
+#undef GEMMI_WRITE_IMPLEMENTATION
+
+#include "chargefw2.h"
+#include "cif.h"
+#include "../structures/molecule_set.h"
+#include "../charges.h"
+#include "../config.h"
+
+
+namespace fs = std::filesystem;
+
+static const std::vector<std::string> atom_site_columns{
+        "group_PDB",
+        "auth_asym_id",
+        "auth_seq_id",
+        "pdbx_PDB_ins_code",
+        "auth_comp_id",
+        "auth_atom_id",
+        "label_alt_id",
+        "type_symbol",
+        "Cartn_x",
+        "Cartn_y",
+        "Cartn_z",
+        "pdbx_PDB_model_num",
+    };
+
+
+class MCRA {
+public:
+    MCRA(const int model,
+         const std::string &chain,
+         const std::string &res_num,
+         const std::string &residue,
+         const std::string &atom)
+        : _model(model), _chain(chain), _res_num(res_num), _residue(residue), _atom(atom)
```

Comment:
Won't compile with GCC:
error: ‘MCRA::_res_num’ will be initialized after [-Werror=reorder]